mods/gen2/random-teams.js

```diff
@@ -91,13 +91,12 @@ class RandomGen2Teams extends RandomGen3Teams {
 
 			// Moves counter.
 			restrictMoves = set.other.restrictMoves;
-			let moves = set.moveset.moves;
-			for (let i = 0; i < moves.length; i++) {
-				if (restrictMoves[moves[i]]) restrictMoves[moves[i]]--;
-				if (restrictMoves['phazing'] && (moves[i] === "roar" || moves[i] === "whirlwind")) {
+			for (const move of set.moveset.moves) {
+				if (restrictMoves[move]) restrictMoves[move]--;
+				if (restrictMoves['phazing'] && ['roar', 'whirlwind'].includes(move)) {
 					restrictMoves['phazing']--;
 				}
-				if (restrictMoves['sleeping'] && (moves[i] === "sleeppowder" || moves[i] === "lovelykiss" || moves[i] === "sing" || moves[i] === "hypnosis" || moves[i] === "spore")) {
+				if (restrictMoves['sleeping'] && ['hypnosis', 'lovelykiss', 'sing', 'sleeppowder', 'spore'].includes(move)) {
```

Comment:
(This `includes` refactor is fine, though; four is definitely the point at which I'd switch to `includes`)